### PR TITLE
common, osd: various cleanups

### PR DESCRIPTION
**Pull request by tchaikov**

### CODE REVIEW

File: src/common/config.cc [hunk 1]
```diff
@@ -863,18 +863,19 @@ int md_config_t::get_val(const std::string &key, char **buf, int len) const
   return _get_val(key, buf,len);
 }
 
-Option::value_t md_config_t::get_val_generic(const std::string &key) const
+const Option::value_t& md_config_t::get_val_generic(const std::string &key) const
```
Comment (adamemerson) [inline src/common/config.cc]: Yay!

File: src/mon/OSDMonitor.cc [hunk 2]
```diff
@@ -226,7 +226,7 @@ void OSDMonitor::create_initial()
     newmap.decode(bl);
     newmap.set_fsid(mon->monmap->fsid);
   } else {
-    newmap.build_simple(g_ceph_context, 0, mon->monmap->fsid, 0);
+    newmap.build_simple(cct, 0, mon->monmap->fsid, 0);
```
Comment (adamemerson) [inline src/mon/OSDMonitor.cc]: Double-yay!

**Comment by tchaikov** [comment]
@adamemerson could you take a look again? i 

- rebased it against master
- added a commit to use `boost::get<>` instead of `static_visitor`

**Comment by adamemerson** [comment]
Looks good to me.

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2017-10-09_18:26:57-rados-wip-kefu-testing-2017-10-10-0049-distro-basic-smithi/
- http://pulpito.ceph.com/kchai-2017-10-10_02:04:03-rados-wip-kefu-testing-2017-10-10-0049-distro-basic-mira/

**Pull request merged**